### PR TITLE
Migrate `./components/widgets/icons/fa_warning_icon.tsx` from Class Component to Function Component 

### PR DESCRIPTION
**Pull request by aditipatelpro**
#### Summary

Migrate `./components/widgets/icons/fa_warning_icon.tsx` from Class Component to Function Component 
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/24760
#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by devinbinnie** [comment]
@aditipatelpro Thank you for the PR! Looks like the tests failing due to outdated snapshots. Can you run `npm run test -- -u` locally and commit the updated snapshots? Thanks again!

**Comment by aditipatelpro** [comment]
> @aditipatelpro Thank you for the PR! Looks like the tests failing due to outdated snapshots. Can you run `npm run test -- -u` locally and commit the updated snapshots? Thanks again!

@devinbinnie, Thanks for your feedback, I have updated the snapshots!

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @aditipatelpro!

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx [hunk 1]
```diff
@@ -10,15 +10,13 @@ type Props = {
...
+    return (
+        <i
+            className={classNames('fa fa-warning', additionalClassName)}
+            title={intl.formatMessage({id: 'generic_icons.warning', defaultMessage: 'Warning Icon'})}
+        />
+    );
+};
 
-export default injectIntl(WarningIcon);
+export default React.memo(injectIntl(WarningIcon));
```
Comment (larkox) [inline webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx:22]: For function components we don`t need `injectIntl`. We can directly use the hook `useIntl`. Can you do the following?
- Remove injectIntl from this line
- Use the `useIntl` hook inside the component
- Remove intl from the props
- Update the snapshots (they are going to break :sweat_smile: )

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just one minor change :smile:

File: webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx (hunk 1 above)
Comment (aditipatelpro) [reply webapp/channels/src/components/widgets/icons/fa_warning_icon.tsx]: @larkox, I have implemented the changes. 😄

**Review by larkox — APPROVED** [review]
Awesome! Thank you very much!

**Comment by larkox** [comment]
@aditipatelpro Thank you for the contribution!

**Pull request merged**